gdx/src/com/badlogic/gdx/math/Rectangle.java

```diff
@@ -217,8 +217,18 @@ public boolean contains (Rectangle rectangle) {
 
 	/** @param r the other {@link Rectangle}
 	 * @return whether this rectangle overlaps the other rectangle. */
-	public boolean overlaps (Rectangle r) {
-		return x < r.x + r.width && x + width > r.x && y < r.y + r.height && y + height > r.y;
+	public boolean overlaps(Rectangle r) {
+		float thisLeft = width >= 0 ? x : x + width;
+		float thisRight = width >= 0 ? x + width : x;
+		float thisButton = height >= 0 ? y : y + height;
```

Comment:
I'm guessing you meant `thisBottom` instead of `thisButton`. This also applies to `thatButton` which should be `thatBottom`.